specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2022-09-01/AppServiceCertificateOrders.json

```diff
@@ -0,0 +1,1801 @@
+          "Create Certificate order": {
+            "$ref": "./examples/CreateAppServiceCertificateOrder.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "delete": {
+        "tags": [
+          "AppServiceCertificateOrders"
+        ],
+        "summary": "Delete an existing certificate order.",
+        "description": "Description for Delete an existing certificate order.",
+        "operationId": "AppServiceCertificateOrders_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "certificateOrderName",
+            "in": "path",
+            "description": "Name of the certificate order.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted certificate order."
+          },
+          "204": {
+            "description": "Certificate order does not exist."
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete App Service Certificate Order": {
+            "$ref": "./examples/DeleteAppServiceCertificateOrder.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "AppServiceCertificateOrders"
+        ],
+        "summary": "Create or update a certificate purchase order.",
+        "description": "Description for Create or update a certificate purchase order.",
+        "operationId": "AppServiceCertificateOrders_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "certificateOrderName",
+            "in": "path",
+            "description": "Name of the certificate order.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateDistinguishedName",
+            "in": "body",
+            "description": "Distinguished name to use for the certificate order.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AppServiceCertificateOrderPatchResource"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "App Service Certificate Order is created.",
```

Comment:
>"App Service Certificate Order is  [](http://example.com/codeflow?start=27&length=34)

Why is it called an 'App Service Certificate Order', and why should it be capitalized? In the API path its not super obvious that a certificateRegistrations RP is implicitly tied to app services, though that appears to be the case....


    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CertificateRegistration/certificateOrders/{certificateOrderName}": {


Is the intention for certificate registrations to only ever be usable with app services? Or to be a more general thing resource you can use with various other resource types? (E.g. what if someone wanted to use custom SSL cert with other Azure services?)

Reply:
For instance, is there any reason I can't use this to manage certificate registrations for generating my SSL certs in a keyvault that I deploy onto a VM with KeyVaultExtension, and selfhost?